src/client/mod.rs

```diff
@@ -320,26 +336,75 @@ impl<'a, T> VaultClient<'a, T>
     }
 
     /// Revoke `VaultClient`'s token. This token can no longer be used.
+    /// Corresponds to [`/auth/token/revoke-self`][token].
+    ///
+    /// ```no_run
```

Comment:
I will update this test to do exactly that one I add a token-generation API, which is next on my list. :-)